### PR TITLE
Added typing annotations to models/video

### PR DESCRIPTION
**Pull request by frgfm**
Following up on #2025, this PR adds missing typing annotations in `models/video`.

Any feedback is welcome!

### CODE REVIEW

**Comment by frgfm** [comment]
Same here @fmassa, the failed test looks unrelated, ready for review :ok_hand:

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:216]: I believe block should be an `nn.Module`. The intention is to allow users pass their own custom blocks

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:217]: `Conv2Plus1D` is also a valid option.

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:219]: This should be optional as per the documentation.

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:271]: Same as above.

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:272]: Same as above.

**Review by datumbox — COMMENTED** [review]
Thanks for the PR.

I spotted few cases where the intended type might be different from what you added. Unfortunately our code was not too clear either in this case. I flagged some of them, let me know your thoughts.

File: torchvision/models/video/resnet.py
Comment (frgfm) [reply torchvision/models/video/resnet.py:219]: The documentation doesn't seem correct then, or is it?
Otherwise I need to specify a default value, should I put `BasicStem`?

File: torchvision/models/video/resnet.py
Comment (frgfm) [reply torchvision/models/video/resnet.py:216]: Alright, but apart from `nn.Module` default class, it needs an `expansion` attribute

Comment (datumbox) [reply torchvision/models/video/resnet.py:216]: You are right, leave as is. :)

File: torchvision/models/video/resnet.py
Comment (datumbox) [reply torchvision/models/video/resnet.py]: Let's fix the documentation instead

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:229]:
```suggestion
            stem (Callable[..., nn.Module]): module class to instantiate the ResNet stem.
```

What you wrote is correct but it seems on Vision we use a [different pattern](https://github.com/pytorch/vision/search?q=%22Callable%5B...%2C+nn.Module%5D%22). Perhaps keep it for consistency reasons?

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:227]: Not sure if this change is worth it but in [other places](https://github.com/pytorch/vision/blob/183a722169421c83638e68ee2d8fc5bd3415c4b4/torchvision/models/mobilenetv3.py#L119-L120) we define it fully: 
```suggestion
            conv_makers (List[Type[Union[Conv3DSimple, Conv3DNoTemporal, Conv2Plus1D]]]): generator function for each layer
```

I leave it up to you on whether you will change this.

**Review by datumbox — DISMISSED** [review]
Thanks @frgfm. Looks good to me. I left few more non-blocking nits. 

Could you also please merge the main branch to your branch (master got remained to main) to see if the failing tests are fixed?

**Comment by frgfm** [comment]
@datumbox Any clue about the reason of this failure?
Apart from if it was because of the out-of-date branch, I haven't modified anything related to the sphinx toctree so the error looks confusing to me :thinking:

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:226]:
```suggestion
            block (Union[BasicBlock, Bottleneck]): resnet building block
```

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:227]:
```suggestion
            conv_makers (List[Type[Union[Conv3DSimple, Conv3DNoTemporal, Conv2Plus1D]]]): generator
```

**Review by datumbox — COMMENTED** [review]
Apparently
```
./torchvision/models/video/resnet.py:223 in private method `__init__`:
        D417: Missing argument descriptions in the docstring (argument(s) conv_makers are missing descriptions in '__init__' docstring)
```

Try these:

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:216]:
```suggestion
        block: Type[Union[BasicBlock, Bottleneck]],
```

File: torchvision/models/video/resnet.py
Comment (datumbox) [inline torchvision/models/video/resnet.py:226]:
```suggestion
            block (Type[Union[BasicBlock, Bottleneck]]): resnet building block
```

**Review by datumbox — APPROVED** [review]
LGTM @frgfm. Thanks for the PR.

I'll merge once the CI is green.

**Pull request merged**
